framework/integration/basic_test.go

```diff
@@ -0,0 +1,85 @@
+// +build k8srequired
+
+package integration
+
+import (
+	"reflect"
+	"testing"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Test_Finalizer_Integration_Basic is a integration test for basic finalizer
+// operations. The test verifies that finalizers are added and removed as
+// expected. It does not cover correct behavior with reconciliation.
+func Test_Finalizer_Integration_Basic(t *testing.T) {
```

Comment:
I wasn't sure about the structure here, as the test is very much streamlined `do A, then B, then C`.